pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctCountThetaSketchAggregationFunction.java

```diff
@@ -1035,6 +1038,24 @@ public Comparable mergeFinalResult(Comparable finalResult1, Comparable finalResu
     return (Long) finalResult1 + (Long) finalResult2;
   }
 
+  @Override
+  public boolean canUseStarTree(Map<String, Object> functionParameters) {
+    Object nominalEntriesParam = functionParameters.get(Constants.THETASKETCH_NOMINAL_ENTRIES);
+    int starTreeNominalEntries;
+
+    // Check if nominal entries values match
+    if (nominalEntriesParam != null) {
+      starTreeNominalEntries = Integer.parseInt(String.valueOf(nominalEntriesParam));
+    } else {
+      // If the functionParameters don't have an explicit nominal entries value set, it means that the star-tree
+      // index was built with
+      // the default value for nominal entries
+      starTreeNominalEntries = CommonConstants.Helix.DEFAULT_THETA_SKETCH_NOMINAL_ENTRIES;
+    }
+    // Check if the query lgK param is less than or equal to that of the StarTree aggregation
```

Comment:
```suggestion
    // Check if the query nominalEntries param is less than or equal to that of the StarTree aggregation
```
nit: looks like this aggregation function directly takes the `K` value rather than `lgK`?

Reply:
Although I suppose it doesn't matter either way since the comparisons are equivalent.